### PR TITLE
Visualization pytest

### PR DESCRIPTION
**Pull request by joeadham**
I made the following changes:
- visualization.py source code was changed to be able to save the images and compare them with the reference images.
- created a directory for the reference images
- created the visualization unit test file
- edited the configs.json as the path of the images in the argument for the visualization_tumour function was hardcoded

If anything could be improved or was done completely wrong, please inform me and I'll update it asap.

### CODE REVIEW

File: tests/test_visualization.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    """
+    contour_ref_path = (
+        "tests/testdata/testcases/visualization_test_photos/contour/volume/"
+    )
+    vs.visualize_tumor(
+        volume=volume_data.numpy(),
+        mask=mask_data,
+        mode="contour",
+        save_path="tests/testdata/data/temp",
```
Comment (Muhammad-Al-Barbary) [inline tests/test_visualization.py:77]: change `"tests/testdata/data/temp"` to `temp_folder_path`

File: tests/test_visualization.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    )
+    for item in os.listdir(temp_folder_path):
+        ref_image = read_image(os.path.join(contour_ref_path, item))
+        test_image = read_image(os.path.join(temp_folder_path, item))
+        images_similarity(test_image, ref_image)
+    delete_temp_images()
+
+
+def test_box():
```
Comment (Muhammad-Al-Barbary) [inline tests/test_visualization.py:86]: high redundancy, use parametrization instead.

File: tests/test_visualization.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        save_path="tests/testdata/data/temp",
+    )
+    for item in os.listdir(temp_folder_path):
+        ref_image = read_image(os.path.join(contour_ref_path, item))
+        test_image = read_image(os.path.join(temp_folder_path, item))
+        images_similarity(test_image, ref_image)
+    delete_temp_images()
+
+
+def test_zoom():
```
Comment (Muhammad-Al-Barbary) [inline tests/test_visualization.py:114]: same as above

File: tests/testdata/testcases/visualization_test_photos/zoom/volume/slice_17_tumor_1.png
Comment (Muhammad-Al-Barbary) [inline tests/testdata/testcases/visualization_test_photos/zoom/volume/slice_17_tumor_1.png:1]: what's wrong here?

**Review by Muhammad-Al-Barbary — COMMENTED** [review]
There is something wrong with the reference visualizations, possibly because of the very low resolution. Make sure you solve it, you don't want your test cases to be based on a buggy implementation.

File: tests/testdata/testcases/visualization_test_photos/box/volume/slice_17_tumor_1.png
[the remark this answers is not in the document]
Comment (joeadham) [reply tests/testdata/testcases/visualization_test_photos/box/volume/slice_17_tumor_1.png:1]: might need to be re-implemted/ debugged later

File: tests/test_visualization.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+
+@pytest.fixture
+def volume_data(volume_path):
+    volume_data = nib.load(volume_path).get_fdata()
+    return  torch.tensor(volume_data)
+
+
+@pytest.fixture
+def mask_data(mask_path):
```
Comment (Muhammad-Al-Barbary) [inline tests/test_visualization.py:36]: Too many unnecessary fixtures, you can replace them with one or two fixtures only.

**Review by Muhammad-Al-Barbary — APPROVED** [review]

**Pull request merged**